src/Bicep.Core/Semantics/Namespaces/NamespaceResolver.cs

```diff
@@ -61,12 +60,17 @@ void TryAddBuiltInNamespace(string @namespace)
             }
 
             TryAddBuiltInNamespace(SystemNamespaceType.BuiltInName);
-            if (sourceFile.FileKind == BicepSourceFileKind.BicepFile)
+
+            if (sourceFile.FileKind == BicepSourceFileKind.BicepFile) // don't register "az" namespace for Bicep Parameters files
             {
-                // don't register "az" namespace for Bicep Parameters files
-                TryAddBuiltInNamespace(AzNamespaceType.BuiltInName);
+                var foundAzProviderDeclaration = sourceFile.ProgramSyntax.Children
+                    .OfType<ImportDeclarationSyntax>()
+                    .FirstOrDefault(x => x.Specification.Name.Equals(AzNamespaceType.BuiltInName, LanguageConstants.IdentifierComparison));
+                if (foundAzProviderDeclaration == null) // If no 'az' provider was explicitly declared, load the embedded one
+                {
+                    TryAddBuiltInNamespace(AzNamespaceType.BuiltInName);
+                }
```

Comment:
Is this necessary? Shouldn't `TryAddBuiltInNamespace` take care of avoiding conflicts already?

Reply:
I guess it does... my bad.